libkpod/diff.go

```diff
@@ -0,0 +1,65 @@
+package libkpod
+
+import (
+	"fmt"
+
+	"github.com/containers/storage/pkg/archive"
+	"github.com/kubernetes-incubator/cri-o/libkpod/image"
+	"github.com/kubernetes-incubator/cri-o/libkpod/layer"
+	"github.com/pkg/errors"
+)
+
+func (c *ContainerServer) GetDiff(from, to string) ([]archive.Change, error) {
+	toLayer, err := c.getLayerID(to)
+	if err != nil {
+		return nil, err
+	}
+	fromLayer := ""
+	if from != "" {
+		fromLayer, err = c.getLayerID(from)
+		if err != nil {
+			return nil, err
+		}
+	}
+	fmt.Println(from, to)
+	fmt.Println(fromLayer, toLayer)
+
+	/*tarStream, err := c.Store().Diff(fromLayer, toLayer, nil)
+	if err != nil {
+		fmt.Println(err)
+	}
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(tarStream)
+	fmt.Println(buf.String())*/ // Get rid of this debugging output when finished
+	return c.Store().Changes(fromLayer, toLayer)
+}
+
+// GetLayerID gets a full layer id given a full or partial id
+// If the id matches a container or image, the id of the top layer is returned
+// If the id matches a layer, the top layer id is returned
+func (c *ContainerServer) getLayerID(id string) (string, error) {
```

Comment:
Not all of it - the expansion of an image name requires logic that's in `FindImage`, and looking up images or containers by name or ID to resolve them to layer IDs isn't something that the `Store` object provides.